github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,405 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceGithubRepositoryRuleset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryRulesetCreateOrUpdate,
+		Read:   resourceGithubRepositoryRulesetRead,
+		// Update: resourceGithubRepositoryRulesetUpdate, // TODO: Implement update, replacement fine for now
+		Delete: resourceGithubRepositoryRulesetDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```

Comment:
The `ForceNew: true,` will be removed once `Update` is implemented, see [previous comment](https://github.com/integrations/terraform-provider-github/pull/1807#discussion_r1271414961)